lib/rules/no-sibling-hooks.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+module.exports = function (context) {
+    var hooks = [ 'before', 'after', 'beforeEach', 'afterEach' ],
+        isUsed = [];
+
+    function newDescribeLayer(describeNode) {
+        return {
+            describeNode: describeNode,
+            before: false,
+            after: false,
+            beforeEach: false,
+            afterEach: false
+        };
+    }
+
+    return {
+        Program: function (node) {
+            isUsed.push(newDescribeLayer(node));
+        },
+
+        CallExpression: function (node) { // eslint-disable-line complexity, max-statements
+            var name;
+            if (node.callee.type !== 'Identifier') {
+              return;
+            }
+
+            name = node.callee.name;
+            if (name === 'describe') {
```

Comment:
Should we also support mocha’s TDD interface which uses `suite` instead of `describe` and maybe also the `describe` alias `context`?

Additionally to that should we also support things like `xdescribe`, `xcontext`, `describe.skip`, `describe.only` etc?